verification/io/other_defs.gobra

```diff
@@ -213,10 +226,9 @@ ghost
 decreases
 pure func dp2_in_ext(s IO_dp2_state, asid IO_as, ifs IO_ifs, pkt IO_pkt2) bool
 
-// TODO: how is this defined?
-// extract_asid
 ghost
 decreases
-pure func (m IO_msgterm) extract_asid() IO_as
-
+pure func (m IO_msgterm) extract_asid() IO_as {
+	return m.MsgTerm_Hash_.MsgTerm_MPair_1.MsgTerm_Key_.Key_macK_
+}
```

Comment:
where does the implementation for this function come from? How is it related to the IO spec in Isabelle?

Reply:
Nvm, I got it. I wonder if we should add preconditions here enforcing which constructors were used for building the msg term.